src/vivarium_csu_hypertension_sdc/components/metrics.py

```diff
@@ -377,3 +379,92 @@ def metrics(self, index, metrics):
 
     def __repr__(self):
         return f"DiseaseCountObserver({self.disease})"
+
+
+class TimeToControlObserver:
+    configuration_defaults = {
+        'metrics': {
+            'time_to_control_observer': {
+                'by_age': True,
+                'by_year': False,
+                'by_sex': True,
+            }
+        }
+    }
+
+    def __init__(self, required_measurement_count: int = 3):
+        self.required_measurement_count = required_measurement_count
+        self.observations = pd.DataFrame()
+
+    @property
+    def name(self):
+        return 'time_to_control_observer'
+
+    def setup(self, builder):
+        self.config = builder.configuration.metrics.time_to_control_observer
+        self.age_bins = get_age_bins(builder)
+        self.population_view = builder.population.get_view(['treatment_start_date',
+                                                            'last_visit_date',
+                                                            'last_visit_type',
+                                                            'high_systolic_blood_pressure_measurement',
+                                                            'age', 'sex'])
+
+        builder.population.initializes_simulants(self.on_initialize_simulants,
+                                                 requires_columns=['sex'],
+                                                 # FIXME: this creates_columns is a hack to get around the resource
+                                                 #  mgr skipping initializers that don't create anything for now
+                                                 creates_columns=['time_to_cotnrol'])
+
+        builder.value.register_value_modifier('metrics', self.metrics)
+        builder.event.register_listener('collect_metrics', self.on_collect_metrics)
+
+    def on_initialize_simulants(self, pop_data):
+        pop = self.population_view.subview(['sex']).get(pop_data.index)
```

Comment:
We might be able to dodge the issue entirely.  We don't need to proactively record this info.  You can keep a df that you build on collect metrics when people enter treatment.

Reply:
I gave up on trying to figure out what was going with the results being in the billions - the types were correct (object not float) - once I got the fix in vivarium and just left it as is since it works.